7896 - readme E2E updates (#7973)
huridocs/uwazidocs: README.md

- **ElasticSearch 7.17.7** https://www.elastic.co/downloads/past-releases/elasticsearch-7-17-6 Please note that ElasticSearch requires Java. Follow the instructions to install the package manually, you also probably need to disable ml module in the ElasticSearch config file:
  `xpack.ml.enabled: false`
- **ICU Analysis Plugin (recommended)** [installation](https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-icu.html#analysis-icu) Adds support for number sorting in texts and solves other language sorting nuances. This option is activated by setting the env var USE_ELASTIC_ICU=true before running the server (defaults to false/unset).

> - **MongoDB 6.0** https://www.mongodb.com/docs/v5.0/installation/ . The MongoDB installation needs to be configured as a Replica Set. It can be a single-node replica set, but Replica Set must be [initialized](https://www.mongodb.com/docs/v6.0/tutorial/deploy-replica-set/). If you have a previous version installed, please follow the instructions on how to [upgrade here](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-standalone/). The new mongosh dependency needs to be added [installation](https://www.mongodb.com/docs/mongodb-shell/).

Reviewer: ```suggestion
- **MongoDB 6.0** https://www.mongodb.com/docs/v6.0/installation/ . The MongoDB installation needs to be configured as a Replica Set. It can be a single-node replica set, but Replica Set must be [initialized](https://www.mongodb.com/docs/v6.0/tutorial/deploy-replica-set/). If you have a previous version installed, please follow the instructions on how to [upgrade here](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-standalone/). The new mongosh dependency needs to be added [installation](https://www.mongodb.com/docs/mongodb-shell/).
```

Reviewer: I am wondering why we don't use Markdown links so we don't need to display the term and the link separate, eg. [**MongoDB 6.0**](https://www.mongodb.com/docs/v6.0/installation/). This would also apply to the other dependencies listed in this section.

Author: Done

Some suites need MongoDB configured in Replica Set mode to run properly. The provided Docker Compose file runs MongoDB in Replica Set mode and initializes the cluster automatically, if you are using your own mongo installation Refer to [MongoDB's documentation](https://www.mongodb.com/docs/manual/tutorial/deploy-replica-set/#initiate-the-replica-set) for more information.

> There are also Cypress components tests. It's recommended that Cypress tests are run with Chrome, or Chrome based browsers.

Reviewer: ```suggestion
There are also Cypress components tests. It's recommended that Cypress tests are run with Chrome or Chrome based browsers.
```

You can run individual tests with the Cypress UI:

```
$ yarn cypress
```

> or you can run tests in headless mode

Reviewer: ```suggestion
or you can run tests in headless mode:
```

```
$ yarn cy-components --browser chrome
```

> ### End to End testing (e2e)

Reviewer: ```suggestion
### End-to-end testing (e2e)
```

> Running end to end tests requires a running Uwazi app. For End-to-End testing, we have a full set of fixtures that test the overall functionality. **It's not advised to run these tests on production environments**, since an incorrectly configured run can have unwanted effects on the production database.

Reviewer: ```suggestion
Running end-to-end tests requires a running Uwazi APP. For End-to-end testing, we have a full set of fixtures that test the overall functionality. **It's not advised to run these tests on production environments**, since an incorrectly configured run can have unwanted effects on the production database.
```

Note that if you already have an instance running, this will likely throw an error of ports already being used. Only one instance of Uwazi may be run in the same port at the same time.

The Uwazi APP needs to run on a specific database and with a specific ElasticSearch index. This is configured via environment variables when starting the application.

#### Running tests with Puppeteer (legacy)

Start UWazi:

```
$ DATABASE_NAME=uwazi_e2e INDEX_NAME=uwazi_e2e yarn hot
```

> On a different console tab, run

Reviewer: ```suggestion
On a different console tab, run:
```

> On a different console tab, run

Reviewer: ```suggestion
On a different console tab, run:
```

```
$ yarn cypress
```

> This will open the Cypress interface where you can select the tests to run. It's recommended that Cypress tests are run with Chrome, or Chrome based browsers.

Reviewer: ```suggestion
This will open the Cypress interface where you can select the tests to run. It's recommended that Cypress tests are run with Chrome or Chrome based browsers.
```

> You can run tests in headless mode, and run individual suites via

Reviewer: ```suggestion
You can run tests in headless mode, and run individual suites via:
```

```
$ yarn cy-e2e --browser chrome --spec path/to/test.cy.ts
```

> Cypress test that use our Information Extraction features need to run Uwazi together with a [dummy service](https://github.com/huridocs/dummy_extractor_services) that mimics the external services needed for the feature.

Reviewer: ```suggestion
Cypress tests that use our Information Extraction features need to run Uwazi together with a [dummy service](https://github.com/huridocs/dummy_extractor_services) that mimics the external services needed for the features.
```

> To run these test you also need to add the following environment variables when running Uwazi:

Reviewer: ```suggestion
To run these tests you also need to add the following environment variables when running Uwazi:
```
